lib/_http_client.js

```diff
@@ -311,11 +311,13 @@ ClientRequest.prototype._implicitHeader = function _implicitHeader() {
                     this[outHeadersKey]);
 };
 
-ClientRequest.prototype.abort = function abort() {
-  if (!this.aborted) {
-    process.nextTick(emitAbortNT.bind(this));
+ClientRequest.prototype.destroy = function destroy(error) {
```

Comment:
In light of https://github.com/nodejs/node/pull/20077, what is the point of destroying with an error if errors after `req.abort` is set should be ignored?

Reply:
I'm trying to get some clarification on that here:

https://github.com/nodejs/node/issues/28698#issuecomment-511593719

> no error event should be emitted after destroy() unless an error was passed to destroy(error) explicitly.

Reply:
`.destroy` in streams accept an optional undocumented callback. Can you add it?

Reply:
As far as I know, it is a bit different. Only one `'error'` event can be emitted but if an error happens between the `destroy()` call and the `'close'` event, it can and is emitted.

Reply:
@mcollina do you have any guidance as to the error? I think @lpinca has a good point but I'm unsure how to resolve it.

Reply:
> In light of #20077, what is the point of destroying with an error if errors after req.abort is set should be ignored?

I think this is complimentary to #20077. The behavior of `.destroy(err)` in streams is that the given error is the last error being emitted. I think we should match the same behavior here.

I'm inclined to have a single PR including this and #20077. I can actually relate more to the goal of having a full `destroy()` here.

---

As for the error issue, the error swallowing should happen only _after_ the error passed to destroy is emitted.

Reply:
> As for the error issue, the error swallowing should happen only after the error passed to destroy is emitted.

This still need to be addressed.

Reply:
@mcollina fixed